src/topology/category/Top/limits.lean

```diff
@@ -43,6 +43,19 @@ def limit_cone (F : J ⥤ Top.{u}) : cone F :=
     naturality' := λ j j' f,
                    continuous_map.coe_inj ((types.limit_cone (F ⋙ forget)).π.naturality f) } }
 
+/--
+An alternative choice of limit cone for a functor `F : J ⥤ Top`
+This is useful when we want to explicitly express limits of topological spaces
+as subspaces of a cartesian product.
+-/
+def alt_limit_cone (F : J ⥤ Top.{u}) : cone F :=
+{ X := { α := { u : Π j : J, F.obj j | ∀ {i j : J} (f : i ⟶ j), F.map f (u i) = u j} },
```

Comment:
Isn't this the same thing as `(F ⋙ forget).sections`? So in fact, how does this construction differ from the `limit_cone` version?

Reply:
The `limit_cone` construction uses an infimum of topologies, whereas this uses a subspace topology. The t2, compact and totally disconnected instances are easier to get with the subspace topology.

Reply:
For example, `is_hausdorff := by {dsimp [Top.alt_limit_cone], apply_instance}` would not work with `Top.limit_cone`.

Reply:
I see, but the topology is still the same (not definitionally), right? So could you instead modify the `limit_cone` construction, or give the subspace topology instance for `(types.limit_cone (F ⋙ forget)).X`. Might it work to just edit the definition of `limit_cone` to this one: I suspect that might not break much since they're provably the same (I think!)

Reply:
Yes, it's the same topology. I can try to change `limit_cone` itself. It looks like @semorrison wrote that code, so perhaps he he can say whether there is a good reason to keep the current definition.

Reply:
I think it's fine to change? I suspect in the past more things may have cared about the definition, but now it's mostly hiding behind propositional `has_limit` gadgets, so there's more flexibility.